### PR TITLE
AGDIGGER-21 - android slave docker image

### PR DESCRIPTION
**Pull request by wtrocki**
## Motivation 

WIP for android docker image.
Slave for android and cordova builds

Creating to open conversation about possible ways to support android versions. 
For now the latest android and cordova are supported.

## TODO 

Cordova support (not required at the moment)
Squashing image laters (moved to stages for visibility and to open conversation about the changes.
Open questions for the team in comments.

### CODE REVIEW

File: docker/android-slave/Dockerfile [hunk 1]
```diff
@@ -16,13 +16,62 @@ RUN yum install -y centos-release-scl-rh && \
...
+# Setup environment
+ENV ANDROID_HOME /opt/android-sdk-linux
+ENV PATH ${PATH}:${ANDROID_HOME}/tools:${ANDROID_HOME}/platform-tools
+
+RUN which adb
+RUN which android
+
+RUN android list targets
+
+# Create emulator
```
Comment (wtrocki) [inline docker/android-slave/Dockerfile]: Should we create emulator at the moment?
I see this as separate implementation to enable and document app testing.
We would need to have some predefined pipeline steps to cover that part (wrapper)

File: docker/android-slave/Dockerfile [hunk 2]
```diff
@@ -16,13 +16,62 @@ RUN yum install -y centos-release-scl-rh && \
...
 
 RUN chown -R 1001:0 $HOME && \
     chmod -R g+rw $HOME
 
-## TODO Extend this to include android build tools
-## See https://github.com/appunite/docker/tree/master/android-java8-r25 
+# Install build dependencies
+RUN yum install -y git wget python curl 
+## Support native builds
+# RUN yum install gcc gcc-c++ make openssl-devel
```
Comment (wtrocki) [inline docker/android-slave/Dockerfile]: We probably need to install native tooling and support native builds. Maybe we can create separate Jenkinsfile for native stuff? Create some low priority jira for that?

File: docker/android-slave/Dockerfile [hunk 3]
```diff
@@ -16,13 +16,62 @@ RUN yum install -y centos-release-scl-rh && \
...
+#                 --force \
+#                 --device "Nexus 5" \
+#                 --name test \
+#                 --target android-24 \
+#                 --abi armeabi-v7a \
+#                 --skin WVGA800 \
+#                 --sdcard 512M
+
+# Install Cordova
+#RUN npm install -g cordova
```
Comment (wtrocki) [inline docker/android-slave/Dockerfile]: Running and installing cordova directly there has many drawbacks and I do not feel we should do it this way. 
Locking up image to one version may be really bad idea. Suggestions:

- npm install -g cordova in Jenkins file (longer builds)
- Separate docker image that is basing on this image with cordova installed (more hassle to update)
- Docker volume with multiple cordova command line tools installed and some support in jenkins file to "mount" proper version on demand (probably to large to be done initially)

**Comment by wtrocki** [comment]
Image pushed to dockerhub for testing.

File: docker/android-slave/Dockerfile [hunk 4]
```diff
@@ -16,13 +16,62 @@ RUN yum install -y centos-release-scl-rh && \
...
+RUN cd /opt && wget --output-document=android-sdk.tgz --quiet https://dl.google.com/android/android-sdk_r24.4.1-linux.tgz && \
+  tar xzf android-sdk.tgz && \
+  rm -f android-sdk.tgz
+
+RUN yum install -y expect expectk
+
+RUN cd /opt && chown -R root.root android-sdk-linux && \
+  chmod -R 777 /opt/tools && \
+  /opt/tools/android-accept-licenses.sh "android-sdk-linux/tools/android update sdk --all --no-ui --filter platform-tools,tools" && \
+  /opt/tools/android-accept-licenses.sh "android-sdk-linux/tools/android update sdk --all --no-ui --filter platform-tools,tools,build-tools-21.0.0,build-tools-21.0.1,build-tools-21.0.2,build-tools-21.1.0,build-tools-21.1.1,build-tools-21.1.2,build-tools-22.0.0,build-tools-22.0.1,build-tools-23.0.0,build-tools-23.0.3,build-tools-24.0.0,build-tools-24.0.1,build-tools-24.0.2,build-tools-24.0.3,build-tools-25.0.0,android-21,android-22,android-23,android-24,android-25,addon-google_apis_x86-google-21,extra-android-support,extra-android-m2repository,extra-google-m2repository,extra-google-google_play_services,sys-img-armeabi-v7a-android-24"
```
Comment (matzew) [inline docker/android-slave/Dockerfile]: This is not OK to do.

That's why for the other digger-build-container we used the 3rd party container

File: docker/android-slave/Dockerfile (hunk 3 above)
Comment (matzew) [inline docker/android-slave/Dockerfile]: `npm install -g cordova` in pipeline files sounds like a good, first step to me. This gives FLEXIBILITY to users/developers. I think folks understand that it means longer builds (if one has used travis before, folks know)

File: docker/android-slave/Dockerfile (hunk 4 above)
Comment (wtrocki) [inline docker/android-slave/Dockerfile]: Yeah. That ticket is just about enabling builds (spike). We have separate ticket to get that extracted. This should not be a big problem.

File: docker/android-slave/Dockerfile (hunk 3 above)
Comment (wtrocki) [inline docker/android-slave/Dockerfile]: Thanks for confirmation. Lets have it that way. Should we create something to cover 3 point?

File: docker/android-slave/Dockerfile (hunk 1 above)
Comment (wtrocki) [inline docker/android-slave/Dockerfile]: Btw. Emulator can be created as part of the jenkins file or library etc.

File: docker/android-slave/Dockerfile (hunk 4 above)
Comment (matzew) [inline docker/android-slave/Dockerfile]: not sure I follow.

but legally we can't do this suggested way

Comment (matzew) [inline docker/android-slave/Dockerfile]: +1

File: docker/android-slave/Dockerfile (hunk 2 above)
Comment (matzew) [inline docker/android-slave/Dockerfile]: Not sure I understand this question

File: docker/android-slave/Dockerfile (hunk 1 above)
Comment (matzew) [inline docker/android-slave/Dockerfile]: Let's not yet worry about the emulator

File: docker/android-slave/Dockerfile (hunk 3 above)
Comment (matzew) [inline docker/android-slave/Dockerfile]: +1

**Comment by matzew** [comment]
ship it for - improving later /cc @odra

File: docker/android-slave/Dockerfile [hunk 5]
```diff
@@ -41,12 +39,10 @@ RUN cd /opt && wget --output-document=android-sdk.tgz --quiet https://dl.google.
...
 
-RUN yum install -y expect expectk
-
 RUN cd /opt && chown -R root.root android-sdk-linux && \
   chmod -R 777 /opt/tools && \
   /opt/tools/android-accept-licenses.sh "android-sdk-linux/tools/android update sdk --all --no-ui --filter platform-tools,tools" && \
-  /opt/tools/android-accept-licenses.sh "android-sdk-linux/tools/android update sdk --all --no-ui --filter platform-tools,tools,build-tools-21.0.0,build-tools-21.0.1,build-tools-21.0.2,build-tools-21.1.0,build-tools-21.1.1,build-tools-21.1.2,build-tools-22.0.0,build-tools-22.0.1,build-tools-23.0.0,build-tools-23.0.3,build-tools-24.0.0,build-tools-24.0.1,build-tools-24.0.2,build-tools-24.0.3,build-tools-25.0.0,android-21,android-22,android-23,android-24,android-25,addon-google_apis_x86-google-21,extra-android-support,extra-android-m2repository,extra-google-m2repository,extra-google-google_play_services,sys-img-armeabi-v7a-android-24"
+  /opt/tools/android-accept-licenses.sh "android-sdk-linux/tools/android update sdk --all --no-ui --filter platform-tools,tools,build-tools-25.0.0,android-25,addon-google_apis_x86-google-21,extra-android-support,extra-google-google_play_services,sys-img-armeabi-v7a-android-24"
```
Comment (aliok) [inline docker/android-slave/Dockerfile]: @matzew @wtrocki mind giving a short update? is this way ok?

**Pull request merged**